pyomo/solvers/plugins/solvers/xpress_direct.py

```diff
@@ -261,6 +265,13 @@ def available(self, exception_flag=True):
         finally:
             xpress.free()
 
+    def _presolve(self, *args, **kwds):
+        # we'll apply the warmstart in _solve_model so the
+        # message "User solution (_) stored" is wrote to the
+        # correct place, i.e., the console or the log or both
+        self._apply_warmstart = kwds.pop("warmstart", False)
+        return super(XpressDirect, self)._presolve(*args, **kwds)
```

Comment:
This is Python2 style - you can do this instead:
```suggestion
        return super()._presolve(*args, **kwds)
```